test/lib/templateFactory.test.js

```diff
@@ -122,4 +128,63 @@ describe('Template Factory', () => {
                 Error, 'No datastore provided to TemplateFactory');
         });
     });
+
+    describe('getTemplate', () => {
+        let config;
+        let expected;
+        let returnValue;
+
+        beforeEach(() => {
+            config = {
+                name: 'testTemplateName',
+                version: '1.0'
+            };
+
+            returnValue = [
+                {
+                    id: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+                    name: 'testTemplateName',
+                    version: '1.0.1',
+                    labels: ['testLabel', 'otherLabel']
+                },
+                {
+                    id: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+                    name: 'testTemplateName',
+                    version: '1.0.2',
+                    labels: ['testLabel', 'otherLabel']
+                },
+                {
+                    id: 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c',
+                    name: 'testTemplateName',
+                    version: '1.0.3',
+                    labels: ['otherLabel']
+                }
```

Comment:
Can we add an entry that is version `2`, to see if it filters it out correctly?
Also, let's make this in random order, since the `rangeKey` doesn't work and might not return them in this ascending order.